key_gen.go

```diff
@@ -84,51 +68,39 @@ func keyForPrefixedStringMaps(prefix string, maps ...map[string]string) string {
 				// Already wrote this key.
 				continue
 			}
-			buf.WriteByte(keyPairSplitter)
+			buf = append(buf, keyPairSplitter)
 		}
 		lastKey = k
 
-		buf.WriteString(k)
-		buf.WriteByte(keyNameSplitter)
+		buf = append(buf, k...)
+		buf = append(buf, keyNameSplitter)
 
 		// Find and write the value for this key. Rightmost map takes
 		// precedence.
 		for j := len(maps) - 1; j >= 0; j-- {
 			if v, ok := maps[j][k]; ok {
-				buf.WriteString(v)
+				buf = append(buf, v...)
 				break
 			}
 		}
 	}
 
-	key := buf.String()
-	keyGenPool.release(buf, keys)
-	return key
+	return buf
 }
 
-func newKeyGenerationPool(size, blen, slen int) *keyGenerationPool {
-	b := NewObjectPool(size)
-	b.Init(func() interface{} {
-		return bytes.NewBuffer(make([]byte, 0, blen))
-	})
-
-	s := NewObjectPool(size)
-	s.Init(func() interface{} {
-		return make([]string, 0, slen)
-	})
-
-	return &keyGenerationPool{
-		bufferPool:  b,
-		stringsPool: s,
-	}
+// keyForPrefixedStringMaps generates a unique key for a prefix and a series
+// of maps containing tags.
+//
+// If a key occurs in multiple maps, keys on the right take precedence.
+func keyForPrefixedStringMaps(prefix string, maps ...map[string]string) string {
+	return string(keyForPrefixedStringMapsAsKey(make([]byte, 0, 128), prefix, maps...))
 }
 
-func (s *keyGenerationPool) release(b *bytes.Buffer, strs []string) {
-	b.Reset()
-	s.bufferPool.Put(b)
-
-	for i := range strs {
-		strs[i] = nilString
+func insertionSort(keys []string) {
```

Comment:
Just to catch any future regressions, should we have a unit test for this function? If it is covered elsewhere that is also cool

Reply:
thanks for the suggestion. added a testcase.